### PR TITLE
community: Vectara Integration Update - Streaming, FCS, Chat, updates to documentation and example notebooks

### PR DESCRIPTION
**Pull request by ofermend**
Thank you for contributing to LangChain!

**Description:** update to the Vectara / Langchain integration to integrate new Vectara capabilities:
- Full RAG implemented as a Runnable with as_rag()
- Vectara chat supported with as_chat()
- Both support streaming response
- Updated documentation and example notebook to reflect all the changes
- Updated Vectara templates

**Twitter handle:** ofermend

**Add tests and docs**: no new tests or docs, but updated both existing tests and existing docs

### CODE REVIEW

**Comment by ofermend** [comment]
Hey @efriis and @baskaryan - just checking about your thoughts on this PR?

**Comment by ofermend** [comment]
@baskaryan @efriis - just updated reranking, added tests, and merged with master so it's up-to-date.
Any further thoughts or feedback so far?

File: libs/community/langchain_community/vectorstores/vectara.py [hunk 1]
```diff
@@ -28,29 +34,30 @@ class SummaryConfig:
...
-    max_results: int = 7
+    max_results: int = 5
     response_lang: str = "eng"
-    prompt_name: str = "vectara-summary-ext-v1.2.0"
+    stream: bool = True
+    prompt_name: str = "vectara-summary-ext-24-05-sml"
 
 
 @dataclass
-class MMRConfig:
```
Comment (efriis) [inline libs/community/langchain_community/vectorstores/vectara.py:37]: I believe this is a breaking change for anyone currently using `MMRConfig` - could we add some kind of backwards compatibility version that allows the old interface as well?

File: libs/community/langchain_community/vectorstores/vectara.py [hunk 2]
```diff
@@ -317,69 +363,101 @@ def add_texts(
             )
         return [doc_id]
 
-    def vectara_query(
```
Comment (efriis) [inline libs/community/langchain_community/vectorstores/vectara.py:320]: any reason someone would have used this before?

File: libs/community/langchain_community/vectorstores/vectara.py [hunk 3]
```diff
@@ -317,69 +363,101 @@ def add_texts(
...
+    def _get_query_body(
```
Comment (efriis) [inline libs/community/langchain_community/vectorstores/vectara.py:366]: thanks for underscore! Will prevent changes to this interface being breaking in future

File: libs/community/langchain_community/vectorstores/vectara.py [hunk 4]
```diff
@@ -66,21 +73,47 @@ class VectaraQueryConfig:
     score_threshold: minimal score threshold for the result.
         If defined, results with score less than this value will be
         filtered out.
-    n_sentence_context: number of sentences before/after the matching segment
+    n_sentence_before: number of sentences before the matching segment
```
Comment (efriis) [inline libs/community/langchain_community/vectorstores/vectara.py:76]: some of these attribute changes are breaking [removals + changed defaults]. Is there a way to do this in a nonbreaking way?

**Review by efriis — COMMENTED** [review]
would it be better to just implement this as a `vectara_v2` retriever instead? There's a few breaking changes in here that will cause issues if anyone is using this currently.

Also - am I correct in my understanding that this only implements streaming for the generic VectaraRAG `Runnable` and not for a retriever?

File: libs/community/langchain_community/vectorstores/vectara.py (hunk 3 above)
Comment (ofermend) [reply libs/community/langchain_community/vectorstores/vectara.py:366]: Not at all. That's why I realized it's best to rename as such.

File: libs/community/langchain_community/vectorstores/vectara.py (hunk 4 above)
Comment (ofermend) [reply libs/community/langchain_community/vectorstores/vectara.py:98]: Great point and thanks for pointing that out. 
I made some changes to make everything backwards compatible. In the process also added test_cases to also test it works in the new form and in the old form, and also updated default values to ensure nothing is a breaking change.

File: libs/community/langchain_community/vectorstores/vectara.py (hunk 2 above)
Comment (ofermend) [reply libs/community/langchain_community/vectorstores/vectara.py:320]: Not at all. I think this one is safe to change.

File: libs/community/langchain_community/vectorstores/vectara.py (hunk 1 above)
Comment (ofermend) [reply libs/community/langchain_community/vectorstores/vectara.py:37]: See your broader comment on breaking change. New update should reflect support for backwards compatibility.

**Comment by ofermend** [comment]
Thanks @efriis for all the great comments and suggestions to making this backwards compatible.
I did a lot of work to make it so today, including adding tests for the same. Please review and let me know if I missed any other point there.
I don't think we need this to be retriever_v2 as these changes address backwards compatibility, but please let me know if you think otherwise.

And yes streaming is only for VectaraRAG. I am unsure how streaming from Vectara could/should work for the retriever alone (which just acts as semantic search) - did you have anything specific in mind here?

**Comment by damianoneill** [comment]
@ofermend, thank you for this PR looking forward to using these updates, glad to see that you made these changes backwards compatible.  

@efriis is there anything additional required in this PR? Apologies, I'm not familiar with the merge process/release schedule, just looking to understand when this would be available in a release?

**Review by baskaryan — APPROVED** [review]

**Pull request merged**